src/plugins/registry/api.go

```diff
@@ -189,13 +190,43 @@ func (registry *Registry) DeleteManifests(ctx *gin.Context) {
 	}
 	account := account_.(auth.Account)
 
-	_, _, err := registry.RegistryAPIDeleteSchemaV2(fmt.Sprintf("%s/%s/manifests/%s", ctx.Param("namespace"), ctx.Param("image"), ctx.Param("reference")), account.Email)
+	var tags []*Tag
+	if ctx.Query("tag") == "" {
+		err = registry.DbClient.Where("namespace = ? AND image = ?", ctx.Param("namespace"), ctx.Param("image")).Find(&tags).Error
+	} else {
+		err = registry.DbClient.Where("namespace = ? AND image = ? AND tag = ? ",
+			ctx.Param("namespace"),
+			ctx.Param("image"),
+			ctx.Query("tag")).Find(&tags).Error
+	}
+
 	if err != nil {
-		err := cranerror.NewError(CodeRegistryManifestDeleteError, err.Error())
-		httpresponse.Error(ctx, err)
+		craneerr := cranerror.NewError(CodeRegistryTagsListError, err.Error())
+		httpresponse.Error(ctx, craneerr)
 		return
 	}
 
+	for _, tag := range tags {
```

Comment:
Make sure delete all tags of a image does not delete a layer belongs to other image.